File: back-end/src/main/java/org/scutsu/materialmanagement/backend/models/InventoryCheck.java
package org.scutsu.materialmanagement.backend.models;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Entity
@Data
public class InventoryCheck {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private User handledBy;

    private Date checkTime;
}

@Entity
@Data
class InventoryCheckNumberedRecord implements Serializable {
    @Id
    @ManyToOne
    private InventoryCheck inventoryCheck;

    @Id
    @ManyToOne
    private MaterialCategory materialCategory;

    @Id
    @Column(length = 20)
    private String number;
}

@Entity
@Data
class InventoryCheckCountedRecord implements Serializable {
    @Id
    @ManyToOne
    private InventoryCheck inventoryCheck;

    @Id
    @ManyToOne
    private MaterialCategory materialCategory;

    @Column(nullable = false)
    private int count;
}

File: back-end/src/main/kotlin/org/scutsu/materialmanagement/backend/models/InventoryCheck.kt
package org.scutsu.materialmanagement.backend.models

import java.util.*
import javax.persistence.*

@Entity
data class InventoryCheck(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,

    @ManyToOne
    val handledBy: User,

    val checkTime: Date
){
}
